app/src/main/java/com/example/tasky/feature_agenda/data/repository/ReminderRepositoryImpl.kt

```diff
@@ -2,91 +2,150 @@ package com.example.tasky.feature_agenda.data.repository
-            Result.Success()
+            Resource.Success()
         } catch(e: HttpException) {
-            Result.Error(e.message ?: "Invalid Response")
+            Resource.Error(message = e.message ?: "Invalid Response", ErrorType.HTTP)
         } catch(e: IOException) {
-            Result.Error(e.message ?: "Couldn't reach server")
+            Resource.Error(e.message ?: "Couldn't reach server", ErrorType.IO)
         }
     }
 
     @RequiresApi(Build.VERSION_CODES.O)
-    override suspend fun getReminder(reminderId: String): Result<AgendaItem.Reminder> {
+    override suspend fun getReminder(reminderId: String): Resource<AgendaItem.Reminder> {
 
         val localReminder = db.reminderDao.getReminder(reminderId.toInt())
 
         localReminder?.let {
-            return Result.Success(localReminder.toReminder())
+            return Resource.Success(localReminder.toReminder())
         }
 
         return try {
 
             val response = api.getReminder(reminderId)
-            val reminder = response.body()?.toReminder() ?: return Result.Error("No such reminder was found!")
+            val reminder = response.body()?.toReminder() ?: return Resource.Error(message = "No such reminder was found!", errorType = ErrorType.OTHER)
 
             val reminderEntity = reminder.toReminderEntity()
             db.reminderDao.upsertReminder(reminderEntity)
-            val newReminder = db.reminderDao.getReminder(reminderId.toInt()) ?: return Result.Error("No such reminder was found!")
+            val newReminder = db.reminderDao.getReminder(reminderId.toInt()) ?: return Resource.Error(message = "No such reminder was found!", errorType = ErrorType.OTHER)
 
-            Result.Success(newReminder.toReminder())
+            Resource.Success(newReminder.toReminder())
         } catch(e: HttpException) {
-            Result.Error(e.message ?: "Invalid Response")
+            Resource.Error(message = e.message ?: "Invalid Response", errorType = ErrorType.HTTP)
         } catch(e: IOException) {
-            Result.Error(e.message ?: "Couldn't reach server")
+            Resource.Error(message = e.message ?: "Couldn't reach server", errorType = ErrorType.IO)
         }
     }
 
-    override suspend fun deleteReminder(reminderId: String): Result<Unit> {
+    override suspend fun deleteReminder(reminder: AgendaItem.Reminder) {
+
+        db.reminderDao.deleteReminder(reminder.reminderId.toInt())
 
-        db.reminderDao.deleteReminder(reminderId.toInt())
+        val jsonReminder = moshi.adapter(AgendaItem.Reminder::class.java).toJson(reminder)
+
+        val inputData = Data.Builder()
+            .putString(ACTION, ActionType.DELETE.toString())
+            .putString(REMINDER, jsonReminder)
+            .build()
+
+        enqueueWorker(
+            workManager = workManager,
+            inputData = inputData,
+            requestBuilder = OneTimeWorkRequestBuilder<ReminderWorker>()
+        )
```

Comment:
I personally wouldn't use a worker for everything, they are rather intended for long-running tasks that need to be executed reliably. In the context of this app, this would be a sync and creating/updating an event (due to up to 10 photos that could be uploaded). 
Creating a reminder or task is something very quick. Adding a different worker for all that will create a lot of overhead and extra complexity in the app which wouldn't be necessary. I personally did it like this:
1. API call triggered in repo
2. If call fails -> save which agenda item couldn't be synced together with CRUD type (whether the item was created, updated or deleted)
3. A periodic worker goes through these locally modified items every 15-30min and tries to sync them with the API